api/envoy/extensions/filters/http/header_mutation/v3/header_mutation.proto

```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.header_mutation.v3;
+
+import "envoy/config/common/mutation_rules/v3/mutation_rules.proto";
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.header_mutation.v3";
+option java_outer_classname = "HeaderMutationProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/header_mutation/v3;header_mutationv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Header mutation filter configuration]
+// [#extension: envoy.filters.http.header_mutation]
+
+// This extension allows for early header mutation by the substitution formatter.
+message HeaderMutation {
+  // The decoder mutations are applied before the request is forwarded to the upstream cluster.
+  repeated config.common.mutation_rules.v3.HeaderMutation decoder_mutations = 1;
+
+  // The encoder mutations are applied before the response is sent to the downstream client.
+  repeated config.common.mutation_rules.v3.HeaderMutation encoder_mutations = 2;
+}
+
+// Empty filter configuration.
```

Comment:
What is this for?

Reply:
Maybe my naming is too ambiguous.

The `HeaderMutation` here is a per route filter proto API. And `HeaderMutationConfig` is a empty filter proto API.

May be using `HeaderMutationPerRoute` and `HeaderMutation` is better?

Reply:
SG with route in the name and explaining this in the comments.